src/OrleansRuntime/Placement/PreferLocalPlacementDirector.cs

```diff
@@ -15,6 +15,10 @@ internal class PreferLocalPlacementDirector : RandomPlacementDirector
         internal override Task<PlacementResult> 
             OnAddActivation(PlacementStrategy strategy, GrainId grain, IPlacementContext context)
         {
+            // if local silo is not active, revert to random placement
+            if (context.LocalSiloStatus != SiloStatus.Active)
```

Comment:
in theory this is behaviorally equivalent to `! context.AllActiveSilos.Contains(context.LocalSilo)`, so you wouldn't need those breaking changes to `IPlacementContext`. I assume you did those because you worried about the efficiency of the alternative, correct?